code/modules/reagents/chemistry/machinery/chem_dispenser.dm

```diff
@@ -387,7 +387,10 @@
 	if(!user)
 		return FALSE
 	if(beaker)
-		user.put_in_hands(beaker)
+		if(in_range(src, user))
```

Comment:
```suggestion
		if(in_range(src, user) && !issilicon(user))
```

For borgs, who do not have a hand with which to hold the beaker, the beaker gets moved under them and is a bit awkward. It would be better if the beaker dropped on top of the machine when a silicon ejects it, regardless if they're a tile away or not..

Reply:
Good idea